developer-extension/src/content-scripts/isolated.ts

```diff
@@ -5,6 +5,18 @@ import { createLogger } from '../common/logger'
 
 const logger = createLogger('content-script-isolated')
 
+chrome.runtime.onMessage.addListener((message) => {
+  if (message && typeof message === 'object' && 'type' in message && message.type === 'settings_updated') {
```

Comment:
Nit: Perhaps this version would work?
```suggestion
  if (message?.type === 'settings_updated') {
```